test/integration/apis/issue.test.ts

```diff
@@ -87,44 +79,101 @@ describe("issue", () => {
         });
 
         it("should request if account is set", async () => {
             issueAPI.setAccount(alice);
             const amount = api.createType("Balance", 1);
             const bobVaultId = api.createType("AccountId", bob.address);
-            const requestResult = await issueAPI.request(amount, bobVaultId);
+            const requestResult = await issueAPI.request(amount);
             txHash = requestResult.hash;
             assert.isDefined(txHash);
         });
 
         it("should consider execution successful if `isExecutionSucessful` returns true", async () => {
-            const { issueId, txId, txBlockHeight, merkleProof, rawTx } = makeExecutionData();
+            const { issueId, txId, merkleProof, rawTx } = makeExecutionData();
             issueAPI.setAccount(alice);
             ImportMock.mockFunction(issueAPI, "isExecutionSucessful", true);
-            const isExecutionCorrect = await issueAPI.execute(issueId, txId, txBlockHeight, merkleProof, rawTx);
+            const isExecutionCorrect = await issueAPI.execute(issueId, txId, merkleProof, rawTx);
             assert.isTrue(isExecutionCorrect);
         });
 
         it("should consider execution failed if `isExecutionSucessful` returns false", async () => {
-            const { issueId, txId, txBlockHeight, merkleProof, rawTx } = makeExecutionData();
+            const { issueId, txId, merkleProof, rawTx } = makeExecutionData();
             issueAPI.setAccount(alice);
             ImportMock.mockFunction(issueAPI, "isExecutionSucessful", false);
-            const isExecutionCorrect = await issueAPI.execute(issueId, txId, txBlockHeight, merkleProof, rawTx);
+            const isExecutionCorrect = await issueAPI.execute(issueId, txId, merkleProof, rawTx);
             assert.isFalse(isExecutionCorrect);
         });
+
+        it("should request and execute issue", async () => {
+            const btcCore = new DefaultBTCCoreAPI("http://0.0.0.0:3002");
+            btcCore.initializeClientConnection("regtest", "0.0.0.0", "rpcuser", "rpcpassword", "18443", "Alice");
+            let initialBalance = await treasuryAPI.balancePolkaBTC(api.createType("AccountId", alice.address));
+            const blocksToMine = 3;
+            keyring = new Keyring({ type: "sr25519" });
+            alice = keyring.addFromUri("//Alice");
+            dave = keyring.addFromUri("//Dave");
+
+            // request issue
+            issueAPI.setAccount(alice);
+            const amountAsBtcString = "0.001";
+            const amountAsSatoshiString = btcToSat(amountAsBtcString);
+            if (amountAsSatoshiString === undefined) {
+                fail();
+            }
+            const amountAsSatoshi = api.createType("Balance", amountAsSatoshiString);
+            const requestResult = await issueAPI.request(amountAsSatoshi);
+            assert.isTrue(requestResult.hash.length > 0);
+
+            // send btc tx
```

Comment:
I would move all of this into a util function that you can share among issue and redeem. Something like
```js
send_data_btc_tx_and_mine(string: data, string: amount, string: recipient)
```